cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -404,13 +404,34 @@ def get_balance(self, pretty_output: bool = True) -> Union[str, float]:
             return f"${balance:,.2f}"
         return balance
 
-    def get_targets(self) -> Dict[str, List[str]]:
-        """Gets a list of available, unavailable, and retired targets.
+    def get_targets(
+        self,
+        simulator: Optional[bool] = None,
+        supports_submit: Optional[bool] = None,
+        supports_submit_qubo: Optional[bool] = None,
+        supports_compile: Optional[bool] = None,
+        available: Optional[bool] = None,
+        retired: Optional[bool] = None,
```

Comment:
definitely like having these written like this, but imo you also had a good point here:
> initial motivation was that if more flags were added on the server side, we wouldn't need to worry as much client-side

wdyt about tacking **kwargs to the end of this to simplify the addition of new server-side filters, while keeping the documentation for known filters?
```suggestion
        retired: Optional[bool] = None,
        **kwargs: bool,
```
then (assuming other suggestions) i don't think we'd even have to change anything in `gss._SuperstaqClient`

Reply:
I think this would be helpful to future-proof, though I guess we would also have to validate the keywords and throw an exception if the filter is not accepted, so I am adding that server-side